src/seedu/addressbook/data/person/Birthday.java

```diff
@@ -0,0 +1,65 @@
+package seedu.addressbook.data.person;
+
+import seedu.addressbook.data.exception.IllegalValueException;
+
+/**
+ * Represents a Person's birthday in the address book.
+ * Guarantees: immutable; is valid as declared in {@link #isValidBirthday(String)}
+ */
+public class Birthday {
+
+    public static final String EXAMPLE = "12-03-1990";
+    public static final String MESSAGE_BIRTHDAY_CONSTRAINTS =
+            "Person birthday in format DD-MM-YYYY where DD is days and MM is months and YYYY is years";
+    public static final String BIRTHDAY_VALIDATION_REGEX = "(((0[1-9]|[12]\\d|3[01])[-](0[13578]|1[02])[-]((1[6-9]|"
+                + "[2-9]\\d)\\d{2}))|((0[1-9]|[12]\\d|30)[-](0[13456789]|1[012])[-]((1[6-9]|[2-9]\\d)\\d{2}))"
+                + "|((0[1-9]|1\\d|2[0-8])[-]02[-]((1[6-9]|[2-9]\\d)\\d{2}))|(29[-]02[-]((1[6-9]|[2-9]\\d)"
+                + "(0[48]|[2468][048]|[13579][26])|((16|[2468][048]|[3579][26])00))))";
```

Comment:
This regex...is extremely developer-unfriendly :o  

It exhibits nice effort on your part, but are you sure this is the simplest regex possible for birthday validation? If it is, you can consider breaking it down into sections like:
```
public static final String BIRTHDAY_DAY_VALIDATION_REGEX = xxx;
public static final String BIRTHDAY_MONTH_VALIDATION_REGEX = xxx;
public static final String BIRTHDAY_YEAR_VALIDATION_REGEX = xxx;
public static final String BIRTHDAY_VALIDATION_REGEX = BIRTHDAY_DAY_VALIDATION_REGEX + BIRTHDAY_MONTH_VALIDATION_REGEX + BIRTHDAY_YEAR_VALIDATION_REGEX;
```

so that it does not appear so intimidating (and difficult to understand!)